### PR TITLE
Document indices in obs/action_set

### PR DESCRIPTION
**Pull request by AntoinePrv**
## Pull request checklist
- [ ] I have opened an issue to discuss the proposed changes: Fix #XX.
- [ ] I have modified/added tests to cover the new changes/features.
- [x] I have modified/added the documentation to cover the new changes/features.
- [x] I have ran the tests, checks, and code formatters.

## Proposed implementation
Documentation for variables indices usage in obseravtions / action_set.
Only the doc-string were modified.

### CODE REVIEW

File: python/ecole/src/ecole/core/dynamics.cpp [hunk 1]
```diff
@@ -87,7 +87,11 @@ void bind_submodule(pybind11::module_ const& m) {
 						Whether the instance is solved.
 						This can happen without branching, for instance if the instance is solved during presolving.
 					action_set:
-						List of branching candidates. Candidates depend on parameters in :py:meth:`__init__`.
+						List of indices of branching candidates variables.
+						Available candidates depend on parameters in :py:meth:`__init__`.
+						Variables indices (values in the ``action_set``) are their postion in the original problem
```
Comment (gasse) [inline python/ecole/src/ecole/core/dynamics.cpp:92]:
```suggestion
						Variable indices (values in the ``action_set``) are their position in the original problem
```

File: python/ecole/src/ecole/core/dynamics.cpp [hunk 2]
```diff
@@ -87,7 +87,11 @@ void bind_submodule(pybind11::module_ const& m) {
 						Whether the instance is solved.
 						This can happen without branching, for instance if the instance is solved during presolving.
 					action_set:
-						List of branching candidates. Candidates depend on parameters in :py:meth:`__init__`.
+						List of indices of branching candidates variables.
```
Comment (gasse) [inline python/ecole/src/ecole/core/dynamics.cpp:90]:
```suggestion
						List of indices of branching candidate variables.
```

File: python/ecole/src/ecole/core/dynamics.cpp [hunk 3]
```diff
@@ -87,7 +87,11 @@ void bind_submodule(pybind11::module_ const& m) {
...
+						Available candidates depend on parameters in :py:meth:`__init__`.
+						Variables indices (values in the ``action_set``) are their postion in the original problem
+						(``SCIPvarGetProbindex``).
+						Variables ordering in the ``action_set`` is arbitrary.
```
Comment (gasse) [inline python/ecole/src/ecole/core/dynamics.cpp:94]:
```suggestion
						Variable ordering in the ``action_set`` is arbitrary.
```

File: python/ecole/src/ecole/core/dynamics.cpp [hunk 4]
```diff
@@ -109,7 +113,11 @@ void bind_submodule(pybind11::module_ const& m) {
 					done:
 						Whether the instance is solved.
 					action_set:
-						List of branching candidates. Candidates depend on parameters in :py:meth:`__init__`.
+						List of indices of branching candidates variables.
```
Comment (gasse) [inline python/ecole/src/ecole/core/dynamics.cpp:116]:
```suggestion
						List of indices of branching candidate variables.
```

File: python/ecole/src/ecole/core/dynamics.cpp [hunk 5]
```diff
@@ -109,7 +113,11 @@ void bind_submodule(pybind11::module_ const& m) {
...
+						Available candidates depend on parameters in :py:meth:`__init__`.
+						Variables indices (values in the ``action_set``) are their postion in the original problem
```
Comment (gasse) [inline python/ecole/src/ecole/core/dynamics.cpp:118]:
```suggestion
						Variable indices (values in the ``action_set``) are their position in the original problem
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 6]
```diff
@@ -94,10 +94,12 @@ void bind_submodule(py::module_ const& m) {
 	)")
 			.def_auto_copy()
 			.def_auto_pickle(std::array{"variable_features", "row_features", "edge_features"})
-			.def_readwrite_xtensor(
-				"variable_features",
-				&NodeBipartiteObs::variable_features,
-				"A matrix where each row is represents a variable, and each column a feature of the variables.")
+			.def_readwrite_xtensor("variable_features", &NodeBipartiteObs::variable_features, R"rst(
+					A matrix where each row is represents a variable, and each column a feature of the variables.
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:98]:
```suggestion
					A matrix where each row represents a variable, and each column a feature of the variable.
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 7]
```diff
@@ -169,10 +171,12 @@ void bind_submodule(py::module_ const& m) {
 	)")
 			.def_auto_copy()
 			.def_auto_pickle(std::array{"variable_features", "constraint_features", "edge_features"})
-			.def_readwrite_xtensor(
-				"variable_features",
-				&MilpBipartiteObs::variable_features,
-				"A matrix where each row represents a variable, and each column a feature of the variables.")
+			.def_readwrite_xtensor("variable_features", &MilpBipartiteObs::variable_features, R"rst(
+					A matrix where each row is represents a variable, and each column a feature of the variables.
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:175]:
```suggestion
					A matrix where each row represents a variable, and each column a feature of the variable.
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 8]
```diff
@@ -219,11 +223,13 @@ void bind_submodule(py::module_ const& m) {
 
 		This observation obtains scores for all LP or pseudo candidate variables at a
 		branch-and-bound node.
-		The strong branching score measures the quality of branching for each variable.
+		The strong branching score measures the quality of branching for each variable (higher is better).
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:226]:
```suggestion
		The strong branching score measures the quality of each variable for branching (higher is better).
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 9]
```diff
@@ -94,10 +94,12 @@ void bind_submodule(py::module_ const& m) {
...
 			.def_auto_copy()
 			.def_auto_pickle(std::array{"variable_features", "row_features", "edge_features"})
-			.def_readwrite_xtensor(
-				"variable_features",
-				&NodeBipartiteObs::variable_features,
-				"A matrix where each row is represents a variable, and each column a feature of the variables.")
+			.def_readwrite_xtensor("variable_features", &NodeBipartiteObs::variable_features, R"rst(
+					A matrix where each row is represents a variable, and each column a feature of the variables.
+
+					Variables are ordered according to their postion in the original problem (``SCIPvarGetProbindex``),
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:100]:
```suggestion
					Variables are ordered according to their position in the original problem (``SCIPvarGetProbindex``),
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 10]
```diff
@@ -169,10 +171,12 @@ void bind_submodule(py::module_ const& m) {
...
 			.def_auto_copy()
 			.def_auto_pickle(std::array{"variable_features", "constraint_features", "edge_features"})
-			.def_readwrite_xtensor(
-				"variable_features",
-				&MilpBipartiteObs::variable_features,
-				"A matrix where each row represents a variable, and each column a feature of the variables.")
+			.def_readwrite_xtensor("variable_features", &MilpBipartiteObs::variable_features, R"rst(
+					A matrix where each row is represents a variable, and each column a feature of the variables.
+
+					Variables are ordered according to their postion in the original problem (``SCIPvarGetProbindex``),
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:177]:
```suggestion
					Variables are ordered according to their position in the original problem (``SCIPvarGetProbindex``),
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 11]
```diff
@@ -219,11 +223,13 @@ void bind_submodule(py::module_ const& m) {
...
 		This observation obtains scores for all LP or pseudo candidate variables at a
 		branch-and-bound node.
-		The strong branching score measures the quality of branching for each variable.
+		The strong branching score measures the quality of branching for each variable (higher is better).
 		This observation can be used as an expert for imitation learning algorithms.
 
 		This observation function extracts an array containing the strong branching score for
-		each variable in the problem which can be indexed by the action set.
+		each variable in the problem.
+		Variables are ordered according to their postion in the original problem (``SCIPvarGetProbindex``),
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:231]:
```suggestion
		Variables are ordered according to their position in the original problem (``SCIPvarGetProbindex``),
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 12]
```diff
@@ -233,59 +239,61 @@ void bind_submodule(py::module_ const& m) {
 		----------
 		pseudo_candidates :
 			The parameter determines if strong branching scores are computed for
-			psuedo-candidate variables if true or LP canidate variables if false.
-			By default psuedo-candidates will be computed.
+			pseudo-candidate variables if true or LP candiate variables if false.
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:242]:
```suggestion
			pseudo candidate variables (when true) or LP candidate variables ( when false).
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 13]
```diff
@@ -233,59 +239,61 @@ void bind_submodule(py::module_ const& m) {
...
-				"features",
-				&Khalil2016Obs::features,
-				"A matrix where each row represents a variable, and each column a feature of the variables.")
-			.def_readonly_static("n_static_features", &Khalil2016Obs::n_static_features)
-			.def_readonly_static("n_dynamic_features", &Khalil2016Obs::n_dynamic_features);
+	)");
+	khalil2016_obs.def_auto_copy()
+		.def_auto_pickle(std::array{"features"})
+		.def_readwrite_xtensor("features", &Khalil2016Obs::features, R"rst(
+			A matrix where each row represents a variable, and each column a feature of the variables.
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:285]:
```suggestion
			A matrix where each row represents a variable, and each column a feature of the variable.
```

File: python/ecole/src/ecole/core/observation.cpp [hunk 14]
```diff
@@ -233,59 +239,61 @@ void bind_submodule(py::module_ const& m) {
...
+
+			Variables are ordered according to their postion in the original problem (``SCIPvarGetProbindex``),
```
Comment (gasse) [inline python/ecole/src/ecole/core/observation.cpp:287]:
```suggestion
			Variables are ordered according to their position in the original problem (``SCIPvarGetProbindex``),

```

**Review by gasse — COMMENTED** [review]
A few minor fixes.

**Comment by gasse** [comment]
@AntoinePrv can you look at those fixes, then re-open and re-merge as needed ? Sorry the review came a bit late.

**Comment by AntoinePrv** [comment]
Thanks for the review! I've fixed it in `latest`.

**Pull request merged**